Src/VimCore/VimBuffer.fs

```diff
@@ -639,6 +645,8 @@ type internal VimBuffer
 
         finally 
             _keyInputEndEvent.Trigger x args
+            if _isClosed && not x.IsProcessingInput then
+                _inputClosedEvent.Trigger x

```

Comment:
This same logic needs to be added to `ProcessBufferedKeyInputs` as it can also drive key strokes.